src/utilities/findBreakingChanges.js

```diff
@@ -278,18 +310,74 @@ export function findFieldsThatChangedType(
           description: `${typeName}.${fieldName} was removed.`,
         });
       } else {
-        // Check if the field's type has changed in the new schema.
-        const oldFieldType = getNamedType(oldTypeFieldsDef[fieldName].type);
```

Comment:
Same deal as the comment in `findArgChanges`. We were previously only comparing the results of `getNamedType` which didn't properly consider changes involving List/NonNull.